src/preamble.js

```diff
@@ -203,6 +203,183 @@ var cwrap, ccall;
     return ret;
   }
 
+  // C calling interface for varargs. Like ccall, but with a varargs specifier
+  // in the last argTypes argument, taking one of two forms:
+  //
+  // 1. a string of the form "[c1c2...]", where cn can be:
+  //   "s" for string, "i" for integer,  "u" for uint, "d" for "double",
+  //   c+ => a single-item repeat pattern if varargs > number of items, or
+  //   (c1c2...)+ => multiple-item repeat pattern, c can be s,i,u,d
+  //
+  // Examples:
+  // "[i]"      => an int arg
+  // "[id]"     => an int arg followed by a double arg
+  // "[d+]"     => repeated double args
+  // "[id+]"    => an int arg followed by repeated double args
+  // "[i(ds)+]" => an int arg, then alternating double and string args
+  //
+  // # pass string, followed by a repeating series of double,int pairs:
+  // Module.ccall_varargs("miniprintf", "null", ["string", "[s(di)+]"], ["%s %f %d %f %d\n", "foo", 1.234, 2, 3.14, -100])
+  //
+  // 2. an array containing 2 arrays of the form [nonrepArr, repArr] where:
+  //   nonrepArr: an array of non-repeating specifiers [c1, c2, ...]
+  //   repArr:    an arrays of repeating specifiers [c1, c2, ...]
+  // where cn specifiers are "s","i","u","d", as above
+  //
+  // # pass string, followed by a repeating series of double,int pairs:
+  // Module.ccall_varargs("miniprintf", "null", ["string", [["s"], ["d", "i"]]], ["%s %f %d %s %f %d %s\n", "foo", 1.234, 2, "foo1", 3.14, -100, "goo1"])
+  //
+  // The second form (the compiled version of the first form) can be used
+  // when ccall_varargs() is called repeatedly with the same varargs spec,
+  // e.g. call_varargs() over all elements in a large 2D array.
+  //
+  ccall_varargs = function ccallVarargsFunc(ident, returnType, argTypes, args, opts) {
```

Comment:
Looks like this should be an anonymous function, namely `ccall_varargs = function (ident, returnType, argTypes, args, opts) {`

Reply:
The use of a function name follows the use of a similar name in the declaration of ccall().

Reply:
I mean, you never use `ccallVarargsFunc`, so I think you don't need it. There is `ccall_varargs` referencing the function already.

Reply:
Understood ... the same situation exists with ccall():
````
ccall = function ccallFunc(ident, returnType, argTypes, args, opts) {
````
I assumed it was added for debugging/profiling, as described by kangax here:

````
https://kangax.github.io/nfe/
````

I have no problem removing it, but if I was the maintainer, I would prefer it match the ccall() declaration one way or another. (Note it's still unclear whether ccall_varargs should be a separate call from call() at all -- see Alon's comment above.)

Reply:
Got it. Closure Compiler will likely remove it anyway.